# 🪵🍁🍂🍄🍗🦃🏈 Big Thanksgiving Dinner Analysis 🏈🦃🍗🍄🍂🍁🪵
**Name:** Jomaris Marin     <br>
**Class:** Python For Data Analytics

In [1]:
# Libraries
import pandas as pd
import numpy as np


In [4]:
# Synthetic Thanksgiving Data
np.random.seed(2310)    # set seed to 2310
n = 50     # set n to 50

# creating the variables first (Age, Dinner, Dessert, Duration)
Age = (np.random.normal(37, 18, n).round(0).clip(5, 90))
Num_Dinner_Servings = np.round(4 - 0.0009 * (Age - 40)**2 + np.random.normal(0, 0.3, n)).clip(1, 5)
Num_Dessert_Servings = np.round(np.clip(2.5 - (Age / 50) + np.random.normal(0, 0.5, n), 0, 3))
Dinner_Duration_Hours =  np.round(1.5 + (Age / 40) + np.random.normal(0, 0.5, n), 1).clip(1, 5)

# creating boolean for whether or not they watched tv (older and younger people more likely to watch)
prob_parade = np.clip(    0.3 + 0.4 * np.exp(-((Age - 15)/20)**2) + 0.4 * np.exp(-((Age - 70)/20)**2),0, 1)
Watched_Parade = np.random.binomial(1, prob_parade)

# creating the data frame using the variables we just made
Thanksgiving = pd.DataFrame({
    'Ages' : Age,
    'Dinner_Servings' : Num_Dinner_Servings,
    'Dessert_Servings' : Num_Dessert_Servings,
    'Duration' : Dinner_Duration_Hours,
    'Macy_Parade' : Watched_Parade
})

Thanksgiving.head()


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
0  33.0              4.0               1.0       2.0            1
1  36.0              4.0               2.0       2.2            0
2  37.0              4.0               2.0       3.0            0
3  31.0              4.0               2.0       2.2            1
4  34.0              5.0               2.0       2.0            1

In [5]:
# Use the `describe` function to get the basic summary statistics
Thanksgiving.describe()


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
count  50.000000        50.000000         50.000000  50.00000    50.000000
mean   39.520000         3.920000          1.500000   2.50400     0.680000
std    14.816207         0.395897          0.646813   0.56531     0.471212
min     9.000000         3.000000          0.000000   1.00000     0.000000
25%    31.250000         4.000000          1.000000   2.20000     0.000000
50%    37.500000         4.000000          1.500000   2.55000     1.000000
75%    50.750000         4.000000          2.000000   2.90000     1.000000
max    72.000000         5.000000          3.000000   3.50000     1.000000

## Instructions:
Use this Thanksgiving dataset to explore patterns in the family dining habits.
You’ll need to _**interpret**_ the data using summary statistics, **not just report them**.

You will be using **only the describe function** to answer your questions

### Questions:

1. Looking at the minimum and maximum values of Age, what do they suggest about who attended this Thanksgiving dinner? <br>
**Answer:** The age range spans from 9 to 72, suggesting a multigenerational gathering that included children, adults, and seniors. <br>

2. Looking at the mean and standard deviation of Dinner_Duration, what do they show about how consistent the mealtime was across family members? <br>
**Answer:** With a mean of 2.5 hours and a standard deviation of 0.57, most guests stayed for a fairly consistent amount of time, with only moderate variation in how long they lingered.<br>

3. If someone stayed for 4.8 hours, would you consider them an outlier? _Use the mean and standard deviation to support your answer (Hint: $mean + 2*std$ OR $IQR=Q3−Q1$ and $Upper Bound=Q3+1.5×IQR$)._ <br>
**Answer:** Yes, the threshold is approximately 3.63 hours. Since 4.8 > 3.63, this guest stayed significantly longer than typical and would be considered an outlier.<br>

4. Compare the mean and median of Dinner_Servings. What could a large difference between them indicate about family member behavior? _(Hint: 50% is half way in the data set)_ <br>
**Answer:** Both the mean (3.92) and median (4.0) are very close, indicating that dinner portions were fairly uniform across the group. A large difference would suggest skewed behavior, such as a few guests eating much more or much less than others.<br>

5. Looking at the max value of Dessert_Servings, what type of guest do you imagine that might be, and what could this tell us about dessert consumption overall?  <br>
**Answer:** The guest who had 3 desserts might be a sweet-toothed enthusiast or a younger attendee. This hints that while dessert was available, most people had fewer servings, making 3 an exception rather than the norm.<br>

6. Suppose the standard deviation of Dinner_Serving is small. What does that suggest about how much everyone ate?  <br>
**Answer:** A small standard deviation (0.40) means most guests ate similar amounts, suggesting a shared meal style or portion control—perhaps everyone was served the same dishes. <br>

7. Which statistic (min, max, mean, std) best helps you find the “light eaters”? Explain how you would identify them and if the minimum makes sense.<br>
**Answer:** The minimum is most useful here. A guest with only 3 servings (when most had 4) could be considered a light eater. This makes sense if they were younger, had dietary restrictions, or arrived late. <br>

8. The range (max − min) of Dinner_Duration is quite large. What could that imply about how the dinner was organized?  <br>
**Answer:** The range of 2.5 hours (3.5 − 1.0) suggests flexible timing—some guests may have arrived early and stayed long, while others popped in briefly. This could reflect staggered arrivals or an open-house-style dining experience.<br>

9. You suspect that older guests tend to eat fewer desserts. Could you use only the `describe` function to explore whether that idea might be true before doing a full analysis, or would we have to include something else?  <br>
**Answer:** No, you’d need like a scatterplot or correlation analysis (.corr) to explore this relationship. describe() only gives separate summaries, not how variables interact. <br>

10. What does the mean tell you about how common it was for people to watch the Macy’s Day Parade?  <br>
**Answer:** A mean of 0.68 implies that about 68% of guests watched the parade, making it a fairly common tradition among attendees.<br>

11. What does the standard deviation suggest about the variation in watching behavior? Was parade-watching behavior fairly consistent across people, or was there a lot of difference between individuals?  <br>
**Answer:** A high standard deviation (0.47) for a binary variable suggests mixed behavior—some watched, some didn’t, indicating that parade-watching wasn’t universal. <br>

12. In one sentence, summarize what kind of Thanksgiving this dataset represents. Focus on what the numbers imply rather than just stating the numbers.  <br>
**Answer:** This Thanksgiving was a warm, multigenerational gathering with consistent meals, varied dessert habits, and a shared—but not universal—interest in the Macy’s Thanksgiving Day Parade. <br>

13. Suggest some visuals that would be ideal for exploring this data and state their purpose. <br>
**Answer:** We could do a histogram of Ages to show generational spread or maybe a boxplot of Dinner_Duration to highlight consistency and outliers.
<br>
